### ISSUE TITLE
Sequence Packing with Wrong attention_mask?

### ISSUE THREAD

**Issue by jjjjohnson** [mit-han-lab/fastrl#14]
https://github.com/mit-han-lab/fastrl/blob/bce3df7a4d46473912e9b81bf47bca419729557f/verl/workers/drafter/eagle_background_trainer.py#L417

Hi @Qinghao-Hu  Looks like the attention_mask is all 1s and Sequence Packing will interfere with other responses within the batch in the attention calculation.... This is strange to me...

In the paper says:
"This technique concatenates multiple variable-length
sequences into a single packed sequence, removing padding
while using attention masks to **preserve sequence integrity**
during parallel processing. "

I doubt this implementation will preserve sequence integrity.
